File: frontend/src/domain/job/value-objects/step-name.ts
/**
 * Step name value object.
 */

export enum StepName {
    FetchMetadata = 'fetch_metadata',
    DownloadVod = 'download_vod',
    ExtractAudio = 'extract_audio',
    Transcribe = 'transcribe',
    ParseSubtitles = 'parse_subtitles',
    SliceSegments = 'slice_segments',
    AnalyzeQuality = 'analyze_quality',
    CreateDataset = 'create_dataset',
    Export = 'export',
}

export const getDisplayName = (stepName: StepName): string => {
    return stepName.replace(/_/g, ' ').replace(/\b\w/g, (l) => l.toUpperCase());
};

export const allSteps = (): readonly StepName[] => {
    return Object.values(StepName);
};

File: backend/streamcraft/domain/job/value_objects/step_name.py
"""Step name value object."""

from enum import Enum


class StepName(str, Enum):
    """Available job steps."""

    FETCH_METADATA = "fetch_metadata"
    DOWNLOAD_VOD = "download_vod"
    EXTRACT_AUDIO = "extract_audio"
    TRANSCRIBE = "transcribe"
    PARSE_SUBTITLES = "parse_subtitles"
    SLICE_SEGMENTS = "slice_segments"
    ANALYZE_QUALITY = "analyze_quality"
    CREATE_DATASET = "create_dataset"
    EXPORT = "export"

    def __str__(self) -> str:
        """String representation."""
        return self.value

    @property
    def display_name(self) -> str:
        """Get human-readable display name."""
        return self.value.replace("_", " ").title()
